server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java

```diff
@@ -183,27 +164,30 @@ public void searchNearestVectors(String field, float[] target, KnnCollector coll
             }
             // when acceptDocs is null due to no doc deleted, we will instantiate a new one that would
             // match all docs to allow timeout checking.
-            final Bits updatedAcceptDocs = acceptDocs == null ? new Bits.MatchAllBits(maxDoc()) : acceptDocs;
-            Bits timeoutCheckingAcceptDocs = new Bits() {
-                private static final int MAX_CALLS_BEFORE_QUERY_TIMEOUT_CHECK = 10;
-                private int calls;
-
-                @Override
-                public boolean get(int index) {
-                    if (calls++ % MAX_CALLS_BEFORE_QUERY_TIMEOUT_CHECK == 0) {
-                        queryCancellation.checkCancelled();
-                    }
-
-                    return updatedAcceptDocs.get(index);
-                }
+            in.searchNearestVectors(field, target, collector, acceptDocs);
```

Comment:
```suggestion
            in.searchNearestVectors(field, target, collector, new TimeOutCheckingBits(acceptDocs));
```
?

Reply:
Uff sorry about that! ... fixed now, moved the comment as well.